src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
```

Comment:
Don't use Objects, Arrays, or any other util class. While `java.util.NoSuchElementException` is not strictly a utility class for arrays, consider if it is necessary in the context of this class or if a more specific exception could be used.